radiotherapyai/metrics/test_dice.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2022 Radiotherapy AI Pty Ltd
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you
+# may not use this file except in compliance with the License. You may
+# obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Testing the Dice metric calculations"""
+
+from typing import TypedDict
+
+import numpy as np
+import shapely.geometry
+
+from . import dice
+
+
+def test_dice_from_polygons():
+    """Compare a range of simple easily calculable dice scores to their
+    shapely based equivalent.
+
+    The expected Dice is 2 * intersection_area / sum_of_areas
+    """
+
+    cases: list[_TestCase] = [
```

Comment:
concave shapes?  A bit more complex, but still possible to have a relatively simple case.

Reply:
Yup, I like it. Fixed in 1d7c8fd